ReportCardParser/src/reportcardparser/AbstractReportCardParser.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package reportcardparser;
+
+import java.io.*;
+/**
+ *
+ * @author bachle
+ */
+public abstract class AbstractReportCardParser implements IReportCardParser {
+        private String filePath; // Path to the report card file 
+        
+    /**
+     * Constructor
+     * 
+     * @param filePath Path to the report card file
+     */
+    public AbstractReportCardParser(String filePath) {
+        this.filePath = filePath;
+    }
+    
+    /**
+     * Opens and closes the file
+     * 
+     * @return  null
+     */
+    @Override
+    public ReportCard parse()
+    {
+        BufferedReader br = null;
+        try {
+            br = new BufferedReader(new FileReader(filePath));
+            
+          return parseFile(br);
+            
+        } catch (IOException ioe) {
+            ioe.printStackTrace(System.out);
+        } finally {
+            try {
+                if (br != null) {
+                    br.close();
+                }
+            } catch (IOException ioe) {
+                System.out.println("Error in closing the BufferedReader");
+                ioe.printStackTrace(System.out);
+            }      
+    }
+        return null;
+    }    
+    abstract ReportCard parseFile(BufferedReader br);
```

Comment:
The abstract class and the concrete classes have exception handling and finally blocks.   If your abstract method was defined as follows:
protected abstract ReportCard parse(BufferedReader br) throws IOException;
Then your concrete implementations would not need to perform the exception handling (allowing it to be handled by the abstract parse method)

Reply:
Your abstract parseFile method should be made protected.